server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/InternalRange.java

```diff
@@ -157,7 +172,9 @@ private static String generateKey(double from, double to, DocValueFormat format)
         public void writeTo(StreamOutput out) throws IOException {
             out.writeString(key);
             out.writeDouble(from);
+            out.writeOptionalDouble(originalFrom);
```

Comment:
You need to add version check on the sending side as well. Otherwise, you will be sending something that another (older) side is not expecting to receive.

Reply:
I thought it was not necessary because:
 - nodes running version 8.1.0 would write the additional fields
 - nodes not running version 8.1.0 would not write the additional fields